### PR TITLE
[Type] Remove explicit template instantiation to restore performances

### PR DESCRIPTION
**Pull request by alxbilger**
______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

**Comment by alxbilger** [comment]
[ci-build][with-all-tests]

**Comment by fredroy** [comment]
adding some details:
this loss of performance has been remarked only when the MSVC flag `/Ob2` is not enabled (meaning it should happen only in Debug (normal) and RelWithDebInfo)

File: Sofa/framework/Type/src/sofa/type/Mat.h [hunk 1]
```diff
@@ -1252,24 +1252,6 @@ constexpr Mat<3,3,real> multTranspose(const Mat<3,3,real>& m1, const Mat<3,3,rea
 }
 
 #if !defined(SOFA_TYPE_MAT_CPP)
```
Comment (fredroy) [inline Sofa/framework/Type/src/sofa/type/Mat.h]: You let the preproc test for a future usage ?

**Review by fredroy — APPROVED** [review]

File: Sofa/framework/Type/src/sofa/type/Mat.h (hunk 1 above)
Comment (alxbilger) [reply Sofa/framework/Type/src/sofa/type/Mat.h]: Actually I was hopping to restore the extern template instantiations while keeping the inline. But I have no idea how to do that (without tuning the compiler options). So I guess I can remove it for now

**Comment by damienmarchal** [comment]
Well, to me this rise an important question... do we really should change code structure/architecture because of compiler specific optimisation details ?

**Pull request merged**